pallets/liquidity-pools-gateway/src/mock.rs

```diff
@@ -124,6 +126,18 @@ impl LPMessage for Message {
 	fn dispute_recovery_message(hash: [u8; 32], router: [u8; 32]) -> Self {
 		Self::DisputeMessageRecovery((hash, router))
 	}
+
+	fn is_forwarded(&self) -> bool {
+		unimplemented!("out of scope")
+	}
+
+	fn unwrap_forwarded(self) -> Option<(Self::Domain, H160, Self)> {
+		unimplemented!("out of scope")
+	}
+
+	fn try_wrap_forward(_: Self::Domain, _: H160, _: Self) -> Result<Self, DispatchError> {
+		unimplemented!("out of scope")
+	}
```

Comment:
I think we can simplify a lot this by having two traits. Note that previous `LpMessage` methods and these new methods are or-exclusively used. This means we can have two traits:

- `LpMessage` as it was, used by the gateway.
- `ForwardableMessage` (or a more, similar, cool name), used by the forwarder.

I think there is no need to merge all message methods under the same trait. This will allow us to remove all `unimplemented!` methods in the two mocks, reducing noise

Reply:
In a more deep thought, ideally we should have (names can be changed):

- `LpMessageSerializer` with serialize/deserialize.
- `LpMessageBatch` with batching methods.
- `LpMessageProof` with proofs methods.
- `LpMessageHash` with `message_hash()`.
- `LpMessageForwarder` with forwarding methods.

Note that all the above traits will NOT be in a hierarchy. They are just behaviors that you can add to a T::Message to deal with.

That way, different entities can deal with different message behavior, i.e. spliting tomorrow the gateway in different pallets, etc.

If agree, I think we can refactor this later. But at least I will split now The `LpMessage` from the `LpMessageForwarder` to simplify the mocks

Reply:
Absolutely agree! Luckily, this change won't influence the outcome of the audit. So we can ship it after the audit feature freeze without decreasing the effectiveness.